src/requests.rs

```diff
@@ -316,7 +318,7 @@ macro_rules! request {
     (
         $destination_request:ident from $source_request:ident
         $( extracts $( $field_name:ident : $field_type:tt ),+ $(,)? )?
-        $( $( and )? groups $( $grouped_field_name:ident ),+ $(,)? )?
+        $( $( and )? groups $( $grouped_field_name:ident $( { $transformation:expr } )? $( $optional:literal )? ),+ $(,)? )?
```

Comment:
I'm adding `"optional"` to flag things as optional.  But the match will match any literal, which is suboptimal.  I'd prefer to match just `optional`, but I don't see any way of doing that.  If I use `$( optional )`, then that would match just `optional`, but then I don't have a way to expand that back out it the transcription below, because it isn't assigned to a metavariable.

Reply:
Could we check at runtime that the token is `optional`? It would be nice to have some kind of check.

Reply:
Or I guess you could have two alternatives, one with `optional` and one without, and give the metavariables different names? I've not done many of these macros so I'm just guessing.

Reply:
@Hywan as original author of the `request!` macro, do you have any suggestions for this?

Reply:
What about simply `optional`?

```rust
$( $( and )? groups $( $grouped_field_name:ident $( { $transformation:expr } )? $( optional )? ),+ $(,)? )?
```

Reply:
Yup, I tried doing `$( optional )?`, but I couldn't figure out how to handle the expansion so that it does something different depending on whether it's specified or not, because it isn't bound to a metavariable.

Reply:
Let's keep things as they are now, I may update this later.